python/friesian/src/bigdl/friesian/feature/utils.py

```diff
@@ -214,6 +214,79 @@ def str_to_list(arg, arg_name):
     return arg
 
 
+def featuretable_to_xshards(tbl, convert_cols=None):
+    from bigdl.friesian.feature import FeatureTable
+    from bigdl.orca.data import SparkXShards
+    from pyspark.ml.linalg import DenseVector
+    import numpy as np
+    from bigdl.orca import OrcaContext
+    shard_size = OrcaContext._shard_size
+    assert isinstance(tbl, FeatureTable), "This function only supports friesian FeatureTable"
+    # TODO: partition < node num
+    if convert_cols is None:
+        convert_cols = tbl.columns
+    if convert_cols and not isinstance(convert_cols, list):
+        convert_cols = [convert_cols]
+
+    schema = tbl.df.schema
+    np_type_dict = dict()
+    for col in convert_cols:
+        assert col in tbl.columns, "Column name " + col + " does not exist in the FeatureTable"
+        feature_type = schema[col].dataType
+        np_type = df_type_to_np_type(feature_type)
+        if np_type:
+            np_type_dict[col] = np_type
+
+    def init_result_dict():
+        return {c: [] for c in convert_cols}
+
+    def generate_output(result):
+        for c in convert_cols:
+            if c in np_type_dict:
+                result[c] = np.asarray(result[c], dtype=np_type_dict[c])
+            else:
+                data = result[c]
+                if len(data) > 0 and isinstance(data[0], DenseVector):
+                    result[c] = np.stack(list(map(lambda vector:
+                                                  vector.values.astype(np.float32), data)))
+                else:
+                    "unsupported field {}".format(schema[c])
```

Comment:
The dtype of DenseVector is np.float64, since
```
    def __init__(self, ar):
        if isinstance(ar, bytes):
            ar = np.frombuffer(ar, dtype=np.float64)
        elif not isinstance(ar, np.ndarray):
            ar = np.array(ar, dtype=np.float64)
        if ar.dtype != np.float64:
            ar = ar.astype(np.float64)
        self.array = ar
```

Reply:
Checking the types of columns in the dataframe first to ensure they are all expected?